### PR TITLE
Fix problems blocking travis from checking the code.

### PR DESCRIPTION
**Pull request by petterreinholdtsen**
This pull request merge the changes from pull request #130 and #131 into one pull requests.

### CODE REVIEW

File: lib/libtsd/tsd_task.c [hunk 1]
```diff
@@ -326,10 +326,14 @@ tsd_task_start(struct tsd_task *t)
...
 		}
-		if (getgid() != getegid())
-			(void)setgid(getgid());
-		if (getuid() != geteuid())
-			(void)setuid(getuid());
+		if (getgid() != getegid() && setgid(getgid()) != 0) {
+			ERROR("failed to reset process group");
+			goto fail;
+		}
+		else if (getuid() != geteuid() && setuid(getuid()) != 0) {
```
Comment (dag-erling) [inline lib/libtsd/tsd_task.c]: Pull up as per [style(9)](https://www.freebsd.org/cgi/man.cgi?query=style&sektion=9), or just get rid of the braces altogether, as they are unnecessary once you remove the incorrect `goto fail`.

File: lib/libtsd/tsd_task.c [hunk 2]
```diff
@@ -326,10 +326,14 @@ tsd_task_start(struct tsd_task *t)
 			if (ret != 0)
 				_exit(1);
 		}
-		if (getgid() != getegid())
-			(void)setgid(getgid());
-		if (getuid() != geteuid())
-			(void)setuid(getuid());
+		if (getgid() != getegid() && setgid(getgid()) != 0) {
+			ERROR("failed to reset process group");
+			goto fail;
```
Comment (dag-erling) [inline lib/libtsd/tsd_task.c]: You are now in the child process. The `fail` target is in the parent code. Using `goto fail` here will result in the child process becoming a new, duplicate tsdfx master process which will compete with its parent (and possibly occupy a task slot in the parent forever). Assuming that `setgid()` or `setuid()` will either never fail or always fail, this change turns tsdfx into a fork bomb that also heavily exercises the file system.

The correct behavior is to just fall through to the `_exit(1)` at the bottom of the child block.

File: lib/libtsd/tsd_task.c [hunk 3]
```diff
@@ -326,10 +326,14 @@ tsd_task_start(struct tsd_task *t)
...
+		}
+		else if (getuid() != geteuid() && setuid(getuid()) != 0) {
+			ERROR("failed to reset process user");
+			goto fail;
+		}
 		(*t->func)(t->ud);
```
Comment (dag-erling) [inline lib/libtsd/tsd_task.c]: This should be in an `else` block as well.

**Review by dag-erling — CHANGES_REQUESTED** [review]

**Comment by petterreinholdtsen** [comment]
I've updated the setuid() part of the patch.

**Pull request merged**